lib/eventsource/eventsource.js

```diff
@@ -0,0 +1,326 @@
+'use strict'
+
+const { pipeline } = require('node:stream')
+const {
+  kEvents,
+  kState
+} = require('./symbols')
+const { getGlobalOrigin } = require('../fetch/global')
+const { webidl } = require('../fetch/webidl')
+const { CONNECTING, OPEN, CLOSED, mimeType, defaultReconnectionTime } = require('./constants')
+const { EventSourceStream } = require('./eventsource-stream')
+
+/**
+ * @typedef {object} EventSourceInit
+ * @property {boolean} [withCredentials] indicates whether the request
+ * should include credentials.
+ */
+
+/**
+ * The EventSource interface is used to receive server-sent events. It
+ * connects to a server over HTTP and receives events in text/event-stream
+ * format without closing the connection.
+ * @extends {EventTarget}
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/EventSource
+ * @api public
+ */
+class EventSource extends EventTarget {
+  #url = null
+  #withCredentials = false
+  #readyState = CONNECTING
+  #lastEventId = ''
+  #connection = null
+  #reconnectionTimer = null
+  #controller = new AbortController()
+
+  /**
+   * Creates a new EventSource object.
+   * @param {string} url
+   * @param {EventSourceInit} [eventSourceInitDict]
+   */
+  constructor (url, eventSourceInitDict) {
```

Comment:
eventSourceInitDict needs its own webidl conversions, here's how it's done for MessageEvent: https://github.com/nodejs/undici/blob/d3d825e21ddea12b893bbf7ba0a4c1e5899c85e4/lib/websocket/events.js#L219

Reply:
`url` should use `webidl.converters.USVString` as well

Reply:
@Uzlopak this isn't resolved afaict

Reply:
I didnt resolve it yet 🤔